scipy/spatial/distance.py

```diff
@@ -144,6 +146,67 @@ def _validate_vector(u, dtype=None):
         raise ValueError("Input vector should be 1-D.")
     return u
 
+def directed_hausdorff(u, v):
+    """
+    Computes the directed Hausdorff distance between N-dimensional
+    point sets using a Euclidean metric.
+    
+    .. versionadded:: 0.19.0
+
+    Parameters
+    ----------
+    u : (M,N) ndarray
+        Input array.
+    v : (O,N) ndarray
+        Input array.
+
+    Returns
+    -------
+    d : double
+        The directed Hausdorff distance between arrays `u` and `v`.
+
+    Notes
+    ----------
+    Uses the early break technique and the random sampling approach described
+    by [Taha2015]_. Although worst-case performance is polynomial (as with the
+    brute force algorithm), this is exceedingly unlikely in practice, and
+    almost-linear time complexity performance can normally be expected for the
+    average case.
+
+    References
+    ----------
+    
+    .. [Taha2015] Taha and Hanbury (2015) An efficient algorithm
+                          for calculating the exact Hausdorff distance.
+                          IEEE Transactions On Pattern Analysis And
+                          Machine Intelligence 37: 2153-63. 
+
+    Examples
+    --------
+    Find the directed Hausdorff distance between
+    two 2-D arrays of coordinates:
+
+    >>> from scipy.spatial.distance import directed_hausdorff
+    >>> u = np.array([(35.0456, -85.2672),
+    ...               (35.1174, -89.9711),
+    ...               (35.9728, -83.9422)])
+    >>> v = np.array([(22.0456, 12.2672),
+    ...               (95.1174, 89.9711),
+    ...               (5.4567, 4.9422)])
```

Comment:
It may be more instructive to take points for which it is clear what the answer should be (easier with ints)